update dependencies for 1.2.3

miccferr opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

miccferr committed:
Update meta.yaml

Try adding `poetry` as host dependency based on [this comment](https://github.com/pypa/flit/issues/461)

miccferr committed:
Merge pull request #1 from miccferr/1_2_3

Update meta.yaml

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

miccferr committed:
Update azure-pipelines-linux.yml

Bump linux version to latest, see [this](https://github.com/actions/virtual-environments/issues/3287)

miccferr committed:
Update meta.yaml

Prioritize `fiona` import to avoid circular dependencies (see [this coment](https://github.com/Toblerity/Fiona/issues/944#issuecomment-806362135) )
